akka-cluster-tools/src/main/scala/akka/cluster/singleton/ClusterSingletonManager.scala

```diff
@@ -686,7 +692,9 @@ class ClusterSingletonManager(singletonProps: Props, terminationMessage: Any, se
       stay
 
     case Event(DelayedMemberRemoved(m), BecomingOldestData(Some(previousOldest)))
-        if m.uniqueAddress == previousOldest =>
+        if m.uniqueAddress == previousOldest && !cluster.isTerminated =>
+      // don't act on DelayedMemberRemoved (starting singleton) if this node is shutting its self down,
```

Comment:
I wonder if we should start ignoring those even earlier, already when `Leaving`?

Reply:
I didn't want to change more than necessary. It may stay in Leaving (not moved to Exiting) if there are other unreachable and then it might be better to let it become singleton instead of not having an active singleton at all.